lib/src/source_visitor.dart

```diff
@@ -108,12 +108,45 @@ class SourceVisitor extends ThrowingAstVisitor {
 
   visitAdjacentStrings(AdjacentStrings node) {
     builder.startSpan();
-    builder.startRule();
-    visitNodes(node.strings, between: splitOrNewline);
+    var rule = new Rule();
+    builder.startRule(rule);
+
+    var chunk = zeroSplit();
+    rule.imply(chunk.rule);
+
+    visit(node.strings.first);
+    var previous = node.strings.first;
+    for (var string in node.strings.skip(1)) {
+      // Always split after a string that ends in a newline. Otherwise, indent a
+      // little bit so it's clear that the next string is a continuation rather
+      // than a new line.
+      if (_endsWithNewline(previous)) {
+        builder.writeWhitespace(Whitespace.nestedNewline);
+      } else {
+        builder.indent();
+        split();
+        builder.unindent();
+      }
```

Comment:
> OK. Can you indent +4 instead of +2? +2 is very rarely used at the expression level, which is where a series of adjacent strings are. Cascades are basically the only place where +2 happens there, because they are special magical snowflakes. Collection literals are +2 as well, but when they split, they split more like blocks than like expressions.

This looks a little odd to me, but maybe that's just because it's not what I'm used to. Done.

> So instead of using `indent()` and `unindent()` here, you'll use `nestExpression()` and `unnest()`.

Done.